### PR TITLE
Solved Do not show unnecessary open and closed in Place Page

### LINKED ISSUE #4196: Do not show unnecessary "open" and "closed" in Place Page
Drawing in green "Closes in … minutes" or in red "opens in … minutes" is enough.

### PR DESCRIPTION
**Pull request by pravalgautam**
Signed-off-by: Praval Gautam <pravalgautam@Pravals-MacBook-Air.local>
Closes #4196

### CODE REVIEW

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 1]
```diff
@@ -165,7 +165,7 @@ final class PlacePagePreviewViewController: UIViewController {
         details = nil;
       }
       
-      setScheduleLabel(state: L("editor_time_open"),
+      setScheduleLabel(state: L(""),
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:168]:
```suggestion
      setScheduleLabel(state: "",
```

Even better is to rewrite setScheduleLabel and remove the first argument.

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 2]
```diff
@@ -202,7 +202,7 @@ final class PlacePagePreviewViewController: UIViewController {
         details = nil;
       }
       
-      setScheduleLabel(state: L("closed_now"),
+      setScheduleLabel(state: L(""),
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:205]: ditto

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 3]
```diff
@@ -232,10 +232,19 @@ final class PlacePagePreviewViewController: UIViewController {
     attributedString.append(stateString);
     if (details != nil)
     {
-      let detailsString = NSAttributedString(string: " • " + details!,
-                                             attributes: [NSAttributedString.Key.font: UIFont.regular14(),
-                                                          NSAttributedString.Key.foregroundColor: UIColor.blackSecondaryText()]);
-      attributedString.append(detailsString);
+      if (placePagePreviewData.schedule.state == .closed){
+        let detailsString = NSAttributedString(string: "" + details!,
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:236]:
```suggestion
        let detailsString = NSAttributedString(string: details!,
```

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 4]
```diff
@@ -232,10 +232,19 @@ final class PlacePagePreviewViewController: UIViewController {
...
+                                               attributes: [NSAttributedString.Key.font: UIFont.regular14(),
+                                                            NSAttributedString.Key.foregroundColor: UIColor.systemRed]);
+        attributedString.append(detailsString);
+        
+      }
+      else if (placePagePreviewData.schedule.state == .open){
+        let detailsString = NSAttributedString(string: "" + details!,
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:243]: ditto

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift [hunk 5]
```diff
@@ -232,10 +232,19 @@ final class PlacePagePreviewViewController: UIViewController {
...
     {
-      let detailsString = NSAttributedString(string: " • " + details!,
-                                             attributes: [NSAttributedString.Key.font: UIFont.regular14(),
-                                                          NSAttributedString.Key.foregroundColor: UIColor.blackSecondaryText()]);
-      attributedString.append(detailsString);
+      if (placePagePreviewData.schedule.state == .closed){
+        let detailsString = NSAttributedString(string: "" + details!,
+                                               attributes: [NSAttributedString.Key.font: UIFont.regular14(),
+                                                            NSAttributedString.Key.foregroundColor: UIColor.systemRed]);
+        attributedString.append(detailsString);
```
Comment (biodranik) [inline iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:239]: Why is this line under the "if" clause?

**Review by biodranik — CHANGES_REQUESTED** [review]
How it looks now?

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift (hunk 1 above)
Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:168]: Screenshot 2023-01-10 at 2 46 56 PM
Screenshot 2023-01-10 at 2 46 44 PM

Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:168]: > Even better is to rewrite setScheduleLabel and remove the first argument.
for this case I can't remove the first argument in setScheduleLabel

  case .allDay:
      setScheduleLabel(state: L("twentyfour_seven"),
                       stateColor: UIColor.systemGreen,
                       details: nil);

File: iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift (hunk 5 above)
Comment (pravalgautam) [reply iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:239]: > Why is this line under the "if" clause?

case .allDay:
setScheduleLabel(state: L("twentyfour_seven"),
stateColor: UIColor.systemGreen,
details: nil);

here the details are nil ,  so in the function setScheduleLabel there is no need to check the other lines that's why using an if clause is helpful

**Comment by biodranik** [comment]
The PR should be polished a bit.

**Comment by pravalgautam** [comment]
> The PR should be polished a bit.

I dont know what  this means

**Comment by pravalgautam** [comment]
> The PR should be polished a bit.

I am not sure what the meaning of this sentence is. Can you please rephrase it?

**Comment by rtsisyk** [comment]
@pravalgautam  could you please address the minor comments above to move forward? Thanks.

**Pull request closed without merging**